### PR TITLE
feat: add dl segmentation code

### PR DESCRIPTION
**Pull request by anuragshukla06**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -131,7 +134,11 @@
     type=str,
     help="Output directory name. If not provided, the current date is used for the directory name.",
 )
-
+parser.add_argument(
+    "--dl_segmentation",
```
Comment (kwatra) [inline main.py]: rename to --dl_mire_detection

File: main.py [hunk 2]
```diff
@@ -131,7 +134,11 @@
     type=str,
     help="Output directory name. If not provided, the current date is used for the directory name.",
 )
-
+parser.add_argument(
+    "--dl_segmentation",
+    action='store_true',
+    help="Flag to opt for DL based segmentation",
```
Comment (kwatra) [inline main.py]: Flag to use dl based mire detection

File: main.py [hunk 3]
```diff
@@ -472,10 +480,31 @@ def generate_topography_maps(
...
+            image_gray_with_channels = np.dstack((image_gray.copy(), np.dstack((image_gray_copy, image_gray_copy))))
+            # generate mask
+            mire_mask = get_mask(image_gray_with_channels)
+            # dump mask
+            image_prefix = image_name.split(".jpg")[0]
+            mask_file_path = self.output + "/" + image_prefix + "/" + image_prefix + "_mire_mask.csv"
+            mire_mask = mire_mask.astype(int)
+            generate_colored_image(mire_mask, self.output + "/" + image_prefix + "/" + image_prefix + "_mire_mask.png")
+            np.savetxt(mask_file_path, mire_mask, delimiter=",", fmt="%d")
+            # segment mires
```
Comment (kwatra) [inline main.py]: I would move this logic to a function inside the segmentation_dl file. Call it detect_mires_in_image().

- Pass it the gray scale image and let it do the logic of stacking channels, etc.
- Also pass the output file name which it will write to.

File: main.py [hunk 4]
```diff
@@ -472,10 +480,31 @@ def generate_topography_maps(
...
+            image_cent_list, image_mp = segment_mires(
```
Comment (kwatra) [inline main.py]: rename this function to detect_mires_from_mask()

File: mire_detection.py [hunk 5]
```diff
@@ -355,16 +364,16 @@ def cleanup_plots_heuristics(r_pixels, start_angle, end_angle, jump, n_mires, ou
...
-    for mire in range(n_mires):
-        r_pixels_temp = []
-        for idx, angle in enumerate(np.arange(start_angle, end_angle, jump)):
-            r_pixels_temp.append(r_pixels_cleaned[idx][mire])
-        plt.plot(np.arange(start_angle, end_angle, jump), r_pixels_temp, ls='-', label=str(mire))
-        r_pixels.append(r_pixels_temp)
-    plt.legend()
-    plt.savefig(output_folder+'/'+image_name+'/plots-modified.png')
-    plt.close()
+    # r_pixels = []
```
Comment (kwatra) [inline mire_detection.py]: What is this change? Seems unrelated to the dl based mire detection?

File: segmentation_dl.py [hunk 6]
```diff
@@ -0,0 +1,216 @@
+import numpy as np
```
Comment (kwatra) [inline segmentation_dl.py]: rename file to mire_detection_dl

**Review by kwatra — CHANGES_REQUESTED** [review]
overall looks good.

- Suggested a minor refactoring of moving most of the logic of dl based mire detection to the segmentation_dl file.
- rename segmentation_dl to mire_detection_dl
- There seems to be some unrelated cleanup changes. Please can we undo those changes and submit that as a separate PR to keep things clean.

File: mire_detection.py (hunk 5 above)
Comment (anuragshukla06) [reply mire_detection.py]: just removed some old plots that are not relevant

File: main.py (hunk 1 above)
Comment (kwatra) [reply main.py]: I think the rename got missed.

**Review by kwatra — APPROVED** [review]
LGTM

One minor comment.
